OnlineStoreTCA/Products/ProductDetails/ProductDetailsViewDomain.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  ProductDetailsViewDomain.swift
+//  OnlineStoreTCA
+//
+//  Created by Mohamed Alouane on 9/10/2022.
+//
+
+import Foundation
+import ComposableArchitecture
+
+struct ProductDetailsDomain{
+    struct State: Equatable {
+        var product: Product
+        var dataLoadingStatus = DataLoadingStatus.notStarted
+       
+        var isLoading: Bool {
+            dataLoadingStatus == .loading
+        }
+    }
+    
+    enum Action: Equatable{
+        case fetchProduct
```

Comment:
I think we don't need to fetch the product again for this demo. Let's keep it simple for now and just pass the data in memory from the selected product.